### ISSUE TITLE
Document supported stats and access log labels

### ISSUE THREAD

**Issue by bianpengyuan** [istio/istio#31112]
Envoy stats and access log are widely used with Istio, but Istio currently does not provide any support for them. It would be great if we can test, document, and support a subset of highly demanded stats and access log labels, so that user's Envoy native telemetry based monitoring won't be broken by Istio upgrade.

cc @douglas-reid @mandarjog @kyessenov @jcetkov

**Comment by mandarjog** [comment]
We broke the stats once because the cluster naming format changed. Are there more examples of the types of breakages that happened?

**Comment by jcetkov** [comment]
the inbound cluster labeling change in 1.8 is an example (not sure if the same you refer to). Now there are interesting metrics that are not available in istio, only in the envoy native, such as circuitbreaking or connections, which we'd like to use.

This ask is to have certain set of those documented and tested, so it doesn't break between releases.

**Comment by ramanujadasu** [comment]
If you provide the more details like design document, what make to do for this issue fix, i will take up this task.

**Comment by kyessenov** [comment]
Envoy stats are documented on Envoy website per-component. The stats are sometimes moved when it makes sense with appropriate deprecation warnings.

**Comment by jcetkov** [comment]
the ask here was to commit to a stable set e.g. the cluster naming as generated by istio stays the same between versions so that user creating a dashboard using those labels doesn't need to revisit/fix their reporting on version upgrade

@kyessenov envoy stats are documented nicely - what istio does to them not so much (the stats_config/stats_tags, naming conventions for clusters etc.)

**Comment by jcetkov** [comment]
fair, however I believe documenting it - even if not committing to it and marking it subject to change - would be helpful (and documenting changes to it, if there are some)

**Comment by kyessenov** [comment]
Agree. In general, having a reference xDS tree (which effectively own the stats) and documenting changes to it is extremely useful for Envoy-level understanding of the system.

/assign @istio/wg-networking-maintainers-pilot

**Comment by jcetkov** [comment]
that goes beyond the original ask - but I agree it makes lots of sense - not just for the telemetry, but for writing `EnvoyFilter`s for functionality not exposed by the native istio API

**Comment by kyessenov** [comment]
@jcetkov The two are coupled. Envoy stats are a reflection on generated xDS tree, which is why we can't do one without the other.
